### DISCUSSION TITLE
Is there a Relationship between Language and Speaker Encoder?

### DISCUSSION THREAD

**Discussion by Lizh-TW** [coqui-ai/TTS#688] [General Q&A] [answered]
I'm try to train a multi-speaker TTS in Chinese.
Is there a relationship between language and speaker encoder?
Embedding projections Chinese / English. (speaker encoder trained in English)
![image](https://user-images.githubusercontent.com/58511222/127609537-ad8d4a38-8438-4956-92e3-19240cdb5ece.png)

**Comment by erogol** [comment]
I'd say yes but @Edresson  is the right person to answer probably.

**Comment by Edresson** [comment, marked as answer]
Hi, @Lizh-TW

I recently trained a speaker encoder with all common voice 6.0 and VoxCeleb 1 + 2 (~60 languages) datasets (OUR in the table) and performed the test on the Multilingual LibriSpeech test dataset, results are in the table below. Also in the Table is the SC-GlowTTS paper Speaker Encoder that was trained in English only. Note that in both models depending on the language there is a large variation (although the variation of "OUR" is smaller). I believe that as speaker verification depends on several aspects (including recording conditions, noise, and the like) it is difficult to say that this task is totally language-dependent (maybe in some languages it influences while others don't). 

In addition, to calculate the EER, the number of speakers influences the result (since the more speakers, the more difficult it is not to have intersections), in this test I did not consider the same number of speakers for all languages, eg: Polish has only 4 speakers, while English has 42. But even considering this, for example, Dutch, which only has 6 speakers, still varies a lot (performing close to English, which has many more speakers).

 Ideally, you need to calculate the EER for your training set and check the values, a low EER will indicate a good separation of the speakers in this set, thus being suitable for training your multi-speaker model.
 
![image](https://user-images.githubusercontent.com/28763586/127858260-40d3e241-aa80-4238-b4aa-a2583154112e.png)

**Reply by erogol** [reply]
the new one looks waaaaaaaayyyyy better than the older model.

**Reply by Edresson** [reply]
> the new one looks waaaaaaaayyyyy better than the older model.

yeah :), my guess is that we can improve even more by changing the number of mel coefficients and the hop_length/win_length (ideally considering more data in fewer frames of the spectrogram, making it easier to the model to learn). But I chose to train the model compatible with the TTS audio config (so that in the future it will be possible to use the speaker encoder as a loss function or train the model together with TTS, like Attentron)
